src/quickpicks/utils/quickPickUtils.ts

```diff
@@ -0,0 +1,251 @@
+ */
+export function createEnhancedQuickPick<T extends QuickPickItemWithValue<any>>(
+  items: T[] | Promise<T[]>,
+  options?: EnhancedQuickPickOptions<T>,
+): Promise<{ quickPick: vscode.QuickPick<T>; selectedItems: T[] }> {
+  const quickPick = vscode.window.createQuickPick<T>();
+  let selectedItems: T[] = [];
+
+  // Set standard options
+  if (options) {
+    quickPick.placeholder = options.placeHolder;
+    quickPick.ignoreFocusOut = options.ignoreFocusOut ?? false;
+    quickPick.title = options.title;
+    quickPick.canSelectMany = options.canSelectMany ?? false;
+    quickPick.matchOnDescription = options.matchOnDescription ?? false;
+    quickPick.matchOnDetail = options.matchOnDetail ?? false;
+    quickPick.buttons = options.buttons ?? [];
+  }
+
+  // Set items (handle promise if needed)
+  if (!(items instanceof Promise)) {
+    quickPick.items = items;
+  } else {
+    // Handle promise asynchronously
+    (async () => {
+      quickPick.busy = true;
+      quickPick.items = await items;
+      quickPick.busy = false;
+
+      // Set default selected items if provided (after items are loaded)
+      if (options?.selectedItems && options.selectedItems.length > 0) {
+        quickPick.selectedItems = options.selectedItems;
+      }
+    })();
+  }
+
+  // Set default selected items if provided and items are already available
+  if (!(items instanceof Promise) && options?.selectedItems && options.selectedItems.length > 0) {
+    quickPick.selectedItems = options.selectedItems;
+  }
+
+  // Set up event handlers
+  if (options?.onSelectionChange) {
+    quickPick.onDidChangeSelection((items: readonly T[]) => {
+      options.onSelectionChange?.(items, quickPick);
+    });
+  }
+
+  if (options?.onActiveItemChange) {
+    quickPick.onDidChangeActive((items: readonly T[]) => {
+      options.onActiveItemChange?.(items[0], quickPick);
+    });
+  }
+
+  if (options?.onItemButtonClicked) {
+    quickPick.onDidTriggerItemButton((event) => {
+      options.onItemButtonClicked?.({
+        button: event.button,
+        item: event.item as T,
+        quickPick,
+      });
+    });
+  }
+
+  if (options?.onButtonClicked) {
+    quickPick.onDidTriggerButton((button) => {
+      options.onButtonClicked?.(button, quickPick);
+    });
+  }
+
+  if (options?.onDidAccept) {
+    quickPick.onDidAccept(() => {
+      options.onDidAccept?.(quickPick);
+    });
+  } else {
+    quickPick.onDidAccept(() => {
+      selectedItems = [...quickPick.selectedItems];
+      quickPick.hide();
+      return;
+    });
+  }
+
+  // Show the QuickPick and return a promise that resolves when it's hidden
+  quickPick.show();
+  return new Promise((resolve) => {
+    quickPick.onDidHide(() => {
+      // If we have pre-selected items and haven't captured any selections yet,
+      // use the pre-selected items
+      if (selectedItems.length === 0 && options?.selectedItems) {
+        selectedItems = [...options.selectedItems];
+      }
```

Comment:
I noticed some strange behavior when clicking the ⚙️ icon (either against the Kafka item or the SR item) where it seems to continue the user flow as if clicking "OK" or hitting "Enter":

https://github.com/user-attachments/assets/643eee3f-8cc4-4930-9739-97e8c392871e

I think removing the section here and only assigning `selectedItems` during `onDidAccept` will fix this and keep the behavior consistent with what's currently on `main`.

Reply:
Thanks for taking care of those other fixes @gnana997! Looks like this is the last part needed before we can merge.